provider/datagen/src/transform/icuexport/collator/mod.rs

```diff
@@ -168,14 +168,12 @@ macro_rules! collation_provider {
                             "collation/{}",
                             self.source.collation_han_database()
                         ))?
-                        .filter_map(|entry| {
-                            entry
-                                .file_stem()
-                                .unwrap()
-                                .to_string_lossy()
-                                .into_owned()
-                                .strip_suffix($suffix)
-                                .map(ToString::to_string)
+                        .filter_map(|mut file_name| {
+                            file_name.truncate(file_name.len() - ".toml".len());
+                            file_name.ends_with($suffix).then(|| {
+                                file_name.truncate(file_name.len() - $suffix.len());
+                                file_name
+                            })
```

Comment:
Thought: Not sure why these changes are in this PR but they seem fine

Reply:
`list` previously returned `PathBuf`s which are annoying to work with. All paths inside our data are UTF-8. Erik is changing this in #2913